ffs/scheduler/internal/sjstore/sjstore.go

```diff
@@ -87,29 +98,17 @@ func (s *Store) Finalize(jid ffs.JobID, st ffs.JobStatus, jobError error, dealEr
 func (s *Store) Dequeue() (*ffs.StorageJob, error) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
-	q := query.Query{Prefix: dsBaseJob.String()}
-	res, err := s.ds.Query(q)
-	if err != nil {
-		return nil, fmt.Errorf("querying datastore: %s", err)
-	}
-	defer func() {
-		if err := res.Close(); err != nil {
-			log.Errorf("closing dequeue query result: %s", err)
-		}
-	}()
-	for r := range res.Next() {
-		var j ffs.StorageJob
-		if err := json.Unmarshal(r.Value, &j); err != nil {
-			return nil, fmt.Errorf("unmarshalling job: %s", err)
-		}
-		_, ok := s.executingCids[j.Cid]
-		if j.Status == ffs.Queued && !ok {
-			j.Status = ffs.Executing
-			if err := s.put(j); err != nil {
+
+	for _, job := range s.queued {
```

Comment:
So the logic is: start from older to newer, and execute the oldest Job with a data Cid that isn't already executing.
(Recall that we only execute on job per data Cid at a time).